tools/hidl/CMakeLists.txt

```diff
@@ -33,23 +33,23 @@ add_custom_command ( OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/hidllex.cpp
     DEPENDS hidllex.l
 )
 
-set (    SRCS 
+set (    SRCS
          ${CMAKE_CURRENT_BINARY_DIR}/hidlgram.cpp
          ${CMAKE_CURRENT_BINARY_DIR}/hidllex.cpp
-         hidl_esp_ng.cpp 
-         hidl_utils.cpp 
-         hidlcomp.cpp 
-         main.cpp 
+         hidl_esp_ng.cpp
+         hidl_utils.cpp
+         hidlcomp.cpp
+         main.cpp
     )
 
-include_directories ( 
+include_directories (
          ${CMAKE_CURRENT_BINARY_DIR}   # for generated .h file
          ${HPCC_SOURCE_DIR}/system/include
          ${CMAKE_CURRENT_SOURCE_DIR}
     )
 
 set_source_files_properties (${CMAKE_CURRENT_BINARY_DIR}/hidlgram.cpp PROPERTIES COMPILE_FLAGS -fno-strict-aliasing)
 
-ADD_DEFINITIONS( -D_CONSOLE)
+ADD_DEFINITIONS( -D_CONSOLE -Wunused-variable)
```

Comment:
The following are the preferred options:

```
  -Wall -Wextra -Wno-switch -Wno-unused-parameter -Werror -Wno-error=delete-non-virtual-dtor
```
so that warnings are reported as errors.  (e.g see CMakeList.txt for jlib)

Reply:
I added the options as indicated. There were a number of additional errors that had to be corrected. Please see hidllex.cpp. In that file a method and it's prototype were commented out since they weren't used. However, there are a number of instances in the code that use the definition that controlled its inclusion. Not knowing this part of the code, I decided to comment it out for now.